### PR TITLE
Attempt at making ssr-config compatible with wp-desktop

### PR DESCRIPTION
**Pull request by samouri**
We recently launched a change to the build system: https://github.com/Automattic/wp-calypso/pull/11352 that breaks wp-desktop.

It made it so that instead of generating a bundle per environment, we only generate a single build. T
here are a couple changes to make here and to wp-desktop to make it compatible.  
The changes necessary for wp-calypso are:
1. send over config in `desktop.jade`
2. ~~switch to node style export/import for `config`. We cannot use es6 modules for `config` because it gets pulled in by wp-desktop's node server~~
3. a recent commit added in an assignment to server/page/index's req.context which doesn't exist in wp-desktop (not sure why).
4. ~~remove config from `sections.js`.  I'm not sure why this is necessary just yet~~
5. Rename `client/lib/config` to `client/lib/create-config` to avoid naming collision with Desktop files.

### Testing
1. make sure that regular wp-calypso functionality isn't broken at all. still boots, etc:
 * `make run`
 * `make test`
2. Load up wp-desktop and tune its claypso to this branch.  then change wp-desktop branch to: https://github.com/Automattic/wp-desktop/pull/275.  if both these branches are checked out then it should build successfully

### CODE REVIEW

File: client/sections.js [hunk 1]
```diff
@@ -7,8 +7,7 @@ const path = require( 'path' );
 /**
  * Internal dependencies
  */
-const config = require( 'config' );
```
Comment (gziolo) [inline client/sections.js]: Good catch :)

File: client/config/index.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 /**
  * Internal dependencies
  */
-import createConfig from 'lib/config';
+//import createConfig from 'lib/config';
```
Comment (gziolo) [inline client/config/index.js]: I think we should remove comments and proceed without ES6 modules.

Comment (samouri) [inline client/config/index.js]: definitely

File: client/sections.js (hunk 1 above)
Comment (samouri) [inline client/sections.js]: I'm not sure if this is a good idea, i may put `config` back.  I think i removed it while i was looking into why `config` was breaking everything

**Comment by samouri** [comment]
To test that this works:
- make sure that regular wp-calypso functionality isn't broken at all. still boots, etc
- load up wp-desktop and tune its claypso to this branch.  then change wp-desktop branch to: https://github.com/Automattic/wp-desktop/pull/275.  if both these branches are checked out then it should build successfully

**Comment by gziolo** [comment]
I decided to rename `client/lib/config` to avoid naming collisions with Desktop.

**Comment by samouri** [comment]
> I decided to rename client/lib/config to avoid naming collisions with Desktop.

Sort of feels like we are running away from a problem, but heck its simpler 👍 .  `create-config` is also a better name for the module

**Pull request merged**